tests/python/gates/basic_gates_test.py

```diff
@@ -595,3 +596,61 @@ def mat(theta: float) -> np.ndarray:
                     phase_11=phase_11,
                     norb=norb,
                 )
+
+
+@pytest.mark.parametrize("norb, spin", ffsim.testing.generate_norb_spin(range(4)))
+def test_apply_fswap_gate_matrix_spinful(norb: int, spin: ffsim.Spin):
+    """Test applying fSWAP gate matrix."""
+    rng = np.random.default_rng()
+
+    mat01 = np.array([[0, 1], [1, 0]])
+
+    phase_00 = 1
+    phase_11 = -1
+
+    for _ in range(3):
```

Comment:
This loop is not required. It's present in the other tests because those sample random numbers, but there's no randomness needed here. For the same reason, you don't need to initialize `rng` (the linter probably complains).